code/game/atoms.dm

```diff
@@ -641,24 +642,24 @@
 
 	if(custom_materials)
 		var/list/materials_list = list()
-		for(var/i in custom_materials)
-			var/datum/material/M = i
-			materials_list += "[M.name]"
+		for(var/material as anything in custom_materials)
+			var/datum/material/current_material = material
+			materials_list += "[current_material.name]"
 		. += "<u>It is made out of [english_list(materials_list)]</u>."
 	if(reagents)
 		if(reagents.flags & TRANSPARENT)
 			. += "It contains:"
 			if(length(reagents.reagent_list))
 				if(user.can_see_reagents()) //Show each individual reagent
-					for(var/datum/reagent/R in reagents.reagent_list)
-						. += "[round(R.volume, 0.01)] units of [R.name]"
+					for(var/datum/reagent/current_reagent in reagents.reagent_list)
```

Comment:
if these are 100% guaranteed to be reagents, put `as anything` for them

Reply:
should be guaranteed